indexing-service/src/main/java/org/apache/druid/indexing/overlord/ImmutableWorkerInfo.java

```diff
@@ -195,13 +222,31 @@ public boolean isValidVersion(String minVersion)
     return worker.getVersion().compareTo(minVersion) >= 0;
   }
 
-  public boolean canRunTask(Task task, double parallelIndexTaskSlotRatio)
+  /**
+   * Determines if a specific task can be executed on the worker based on
+   * various capacity, custom limits, and availability conditions.
+   * <p>
+   * returns true only if:
+   * <ul>
+   *   <li>The worker has sufficient capacity to handle the task.</li>
+   *   <li>The task is of the parallel index type and can operate within the permitted ratio of slots designated for that task type.</li>
+   *   <li>The the task can run under custom-defined limits for its type,
+   *   such as a maximum number of tasks allowed or a ratio of slots the task type can occupy.</li>
+   *   <li>The availability group of the task is currently available.</li>
+   * </ul>
+   */
+  public boolean canRunTask(Task task, WorkerTaskRunnerConfig config)
   {
-    return (worker.getCapacity() - getCurrCapacityUsed() >= task.getTaskResource().getRequiredCapacity()
-            && canRunParallelIndexTask(task, parallelIndexTaskSlotRatio)
-            && !getAvailabilityGroups().contains(task.getTaskResource().getAvailabilityGroup()));
+    return (hasSufficientWorkerCapacity(task)
+            && canRunParallelIndexTask(task, config.getParallelIndexTaskSlotRatio())
+            && canRunTaskBasedOnCustomLimit(task, config.getTaskSlotLimits(), config.getTaskSlotRatios())
+            && isAvailabilityGroupAvailable(task));
   }
 
+
+  /**
+   * To be removed in future as it could be fully covered with the limits map.
+   */
```

Comment:
I don't think this is needed, especially since we have not deprecated the old property yet.

Reply:
Removed